src/main/java/mekanism/common/content/matrix/SynchronizedMatrixData.java

```diff
@@ -1,65 +1,274 @@
 package mekanism.common.content.matrix;
 
+import io.netty.buffer.ByteBuf;
 import java.util.HashSet;
 import java.util.Set;
 import mekanism.api.Coord4D;
+import mekanism.api.TileNetworkList;
 import mekanism.common.multiblock.SynchronizedData;
 import mekanism.common.tile.TileEntityInductionCell;
+import mekanism.common.tile.TileEntityInductionProvider;
+import mekanism.common.util.MekanismUtils;
 import net.minecraft.item.ItemStack;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.NonNullList;
 import net.minecraft.world.World;
 
+//TODO: Do something better for purposes of double precision such as BigInt
 public class SynchronizedMatrixData extends SynchronizedData<SynchronizedMatrixData> {
 
-    public NonNullList<ItemStack> inventory = NonNullList.withSize(2, ItemStack.EMPTY);
+    private NonNullList<ItemStack> inventory = NonNullList.withSize(2, ItemStack.EMPTY);
+    private Set<Coord4D> providers = new HashSet<>();
+    private Set<Coord4D> cells = new HashSet<>();
+    private double remainingOutput;
+    private double remainingInput;
+    private double lastOutput;
+    private double lastInput;
 
-    public Set<Coord4D> cells = new HashSet<>();
+    private double cachedTotal;
+    private double transferCap;
+    private double storageCap;
 
-    public Set<Coord4D> providers = new HashSet<>();
-
-    public double remainingInput;
-    public double lastInput;
-
-    public double remainingOutput;
-    public double lastOutput;
-
-    public double clientEnergy;
-    public double storageCap;
-    public double transferCap;
+    private int clientProviders;
+    private int clientCells;
 
     @Override
     public NonNullList<ItemStack> getInventory() {
         return inventory;
     }
 
-    public double getEnergy(World world) {
-        double ret = 0;
+    public void setInventory(NonNullList<ItemStack> inventory) {
+        this.inventory = inventory;
+    }
+
+    public void addCell(Coord4D coord, TileEntityInductionCell cell) {
+        //As we already have the two different variables just pass them instead of accessing world to get tile again
+        cells.add(coord);
+        storageCap += cell.tier.getMaxEnergy();
+        cachedTotal += cell.getEnergy();
+    }
+
+    public void addProvider(Coord4D coord, TileEntityInductionProvider provider) {
+        providers.add(coord);
+        transferCap += provider.tier.getOutput();
+    }
+
+    public double getEnergyPostQueue() {
+        //The reason we do remainingOutput - remainingInput when it logically appears that
+        // it should be the other way around, is because in reality our value is
+        // (transferCap - remainingInput) - (transferCap - remainingOutput)
+        // which simplifies to remainingOutput - remainingInput
+        // This is because remainingInput and remainingOutput go down if we have
+        // the corresponding one queued.
+        return cachedTotal - remainingInput + remainingOutput;
```

Comment:
Unless I'm being dense, that comment doesn't match the variables used in the calculation?

Reply:
The comment is explaining why it is `cachedTotal - remainingInput + remainingOutput` instead of being `cachedTotal + remainingInput - remainingOutput` (what I originally thought it should be).

Given what we want to return is `cachedTotal + queuedReceiving - queuedSending`. I may look into it a bit more before I merge and cleanup the variables *slightly* so that it has a bit better precision by storing the queued receiving/sending in their own variable that is `> 0` and then merge it for the method for the remaining limit.